src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -842,12 +842,33 @@ CHIP_ERROR ConnectivityManagerImpl::ProvisionWiFiNetwork(const char * ssid, cons
     GError * err    = nullptr;
     GVariant * args = nullptr;
     GVariantBuilder builder;
+    gboolean result;
 
     // Clean up current network if exists
     if (mWpaSupplicant.networkPath)
     {
-        g_object_unref(mWpaSupplicant.networkPath);
-        mWpaSupplicant.networkPath = nullptr;
+        GError * error = nullptr;
```

Comment:
can't you use `err` here?

Reply:
if result == true and error is not null (which is highly unlikely), This temporary variable 'error'  needs to be freed after use. So I  prefer not reuse err in the outer space for local error handling.